rotkehlchen/chain/ethereum/names.py

```diff
@@ -67,56 +67,135 @@ def search_for_addresses_names(
-                cursor=global_cursor,
-                addresses=addresses,
-            )
-        private_addressbook = db_addressbook.get_addressbook_entries(
+            for name_source in prioritized_name_source:
+                fetcher = self._fetchers.get(name_source)
+                if not fetcher:
+                    raise NotImplementedError(
+                        f"address name fetcher for '{name_source}' is not implemented")
+
+                name = fetcher(self._db, address)
+                if not name:
+                    continue
+                top_prio_names[address] = name
+                break
+
+        return top_prio_names
+
+
+def blockchain_address_to_name(db: DBHandler, address: ChecksumEvmAddress) -> str:
+    with db.conn.read_ctx() as cursor:
+        accounts_data = db.get_blockchain_account_data(
             cursor=cursor,
-            addresses=addresses,
+            blockchain=SupportedBlockchain.ETHEREUM,
         )
-        addresses_set = set(addresses)
-        labels = {}
-        accounts_data = database.get_blockchain_account_data(
+
+        for account in accounts_data:
+            if account.address == address and account.label:
+                return account.label
+
+    return ""
+
+
+def private_addressbook_address_to_name(db: DBHandler, address: ChecksumEvmAddress) -> str:
+    db_addressbook = DBAddressbook(db)
+    with db.conn.read_ctx() as cursor:
+        addressbook_entries = db_addressbook.get_addressbook_entries(
             cursor=cursor,
-            blockchain=SupportedBlockchain.ETHEREUM,
+            addresses=[address],
         )
+        for addressbook_entry in addressbook_entries:
+            if addressbook_entry.address == address:
+                return addressbook_entry.name
+
+    return ""
+
+
+def global_addressbook_address_to_name(db: DBHandler, address: ChecksumEvmAddress) -> str:
+    db_addressbook = DBAddressbook(db)
+    with db_addressbook.read_ctx(AddressbookType.GLOBAL) as global_cursor:
+        addressbook_entries = db_addressbook.get_addressbook_entries(
+            cursor=global_cursor,
+            addresses=[address],
+        )
+        for addressbook_entry in addressbook_entries:
+            if addressbook_entry.address == address:
+                return addressbook_entry.name
+
+    return ""
 
-    for account in accounts_data:
-        if account.address in addresses_set:
-            labels[account.address] = account.label
 
-    result_mappings: Dict[ChecksumEvmAddress, str] = {}
+def hardcoded_address_to_name(_: DBHandler, address: ChecksumEvmAddress) -> str:
+    constant_name = ETHADDRESS_TO_KNOWN_NAME.get(address, "")
+    return constant_name
+
+
+def token_mappings_address_to_name(_: DBHandler, address: ChecksumEvmAddress) -> str:
+    token_mappings = GlobalDBHandler().get_tokens_mappings(addresses=[address])
+
+    for _addr, name in token_mappings.items():
+        return name
+
+    return ""
```

Comment:
1. Add a docstring.
2. You are asking for a single address and get the mapping for this address. Why are you iterating the mapping and return the first thing you get?

```suggestion
    token_mappings = GlobalDBHandler().get_tokens_mappings(addresses=[address])
    return token_mappings.get(address, None)
```